### PR TITLE
Glm term entry

### LINKED ISSUE #5639: [Term Entry] Python statsmodels: GLM
### Reason/inspiration (optional)

We would like a new entry on the `GLM` term under Python. The entry should go in a new file under `docs/content/python/concepts/statsmodels/terms/glm/glm.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by danialkhan02**
### Description

Created a new term entry for Generalized Linear Models (GLM) under Python's statsmodels library. The entry includes an introduction to GLM concepts, syntax documentation with available family distributions, and practical examples using both the iris dataset and a house price prediction scenario. The content aims to help users understand and implement GLM in Python using the statsmodels package.

### Issue Solved

Closes #5639

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/python/concepts/statsmodels/terms/glm/glm.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+```python
+import statsmodels.api as sm
+
+# Fit the GLM model
+model = sm.GLM(y,                    # Dependent variable
+               X,                    # Independent variables
+               family=sm.families.Family())  # Distribution family
+results = model.fit()
+```
```
Comment (SaviDahegaonkar) [inline content/python/concepts/statsmodels/terms/glm/glm.md:33]:
```suggestion
```pseudo
import statsmodels.api as sm

# Fit the GLM model
# Here `y` is the dependent variable and `X` is the independent variable
model = sm.GLM(y, X, family=sm.families.Family())  # Base class for Distribution family
results = model.fit()
```
```
The syntax must always be included inside the ```pseudo``` blocks.

File: content/python/concepts/statsmodels/terms/glm/glm.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+# Print summary
+print(results.summary())
+
+# Make a prediction for a flower with sepal length=5.0 and sepal width=3.5
+new_flower = np.array([1, 5.0, 3.5])  # include constant term
+prediction = results.predict(new_flower)
+print("\nPredicted petal length:", prediction[0])
+```
+
+Here is the output of the above code:
```
Comment (SaviDahegaonkar) [inline content/python/concepts/statsmodels/terms/glm/glm.md:78]:
```suggestion
Here is the output of the code:
```
Avoid using directional words like `above`, `below`, etc.

File: content/python/concepts/statsmodels/terms/glm/glm.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+==============================================================================
+
+Predicted petal length: 1.6680197099857788
+```
+
+Click here for more examples: [statsmodels GLM documentation](https://www.statsmodels.org/stable/glm.html)
+
+## Codebyte Example
+
+Here's a practical example using GLM to predict house prices:
```
Comment (SaviDahegaonkar) [inline content/python/concepts/statsmodels/terms/glm/glm.md:107]:
```suggestion
Here's an codebyte example that demonstrates the use of  GLM to predict house prices:
```

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hey Author, 
I had a look at your file. Please make the suggested changes asap. Thank you for contributing to Codecademy Docs!

Thanks,
Savi

**Comment by danialkhan02** [comment]
Hi @SaviDahegaonkar thanks for all the comments! I addressed all the comments in my latest commit, let me know if everything else is fine. Thanks!

**Comment by mamtawardhani** [comment]
Hey @SaviDahegaonkar could you check this PR on priority. The author has made the changes, could you push it for the second review asap

File: content/python/concepts/statsmodels/terms/glm/glm.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+- [`sm.families.Poisson()`](https://www.statsmodels.org/stable/generated/statsmodels.genmod.families.family.Poisson.html): For count data
+- [`sm.families.InverseGaussian()`](https://www.statsmodels.org/stable/generated/statsmodels.genmod.families.family.InverseGaussian.html): For positive continuous data with inverse Gaussian distribution
+- [`sm.families.NegativeBinomial()`](https://www.statsmodels.org/stable/generated/statsmodels.genmod.families.family.NegativeBinomial.html): For overdispersed count data
+- [`sm.families.Tweedie()`](https://www.statsmodels.org/stable/generated/statsmodels.genmod.families.family.Tweedie.html): For compound Poisson distributions
+
+Source: [statsmodels GLM documentation](https://www.statsmodels.org/stable/_modules/statsmodels/genmod/generalized_linear_model.html#GLM)
+
+## Example
+
+Here's an example of fitting a GLM using the famous `iris` dataset to predict petal length.
```
Comment (SaviDahegaonkar) [inline content/python/concepts/statsmodels/terms/glm/glm.md:49]:
```suggestion
Here's an example of fitting a GLM using the famous `iris` dataset to predict petal length:
```

File: content/python/concepts/statsmodels/terms/glm/glm.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+model = sm.GLM(y, X, family=sm.families.Gaussian())
+results = model.fit()
+
+# Print summary
+print(results.summary())
+
+# Make a prediction for a flower with sepal length=5.0 and sepal width=3.5
+new_flower = np.array([1, 5.0, 3.5])  # include constant term
+prediction = results.predict(new_flower)
+print("\nPredicted petal length:", prediction[0])
```
Comment (SaviDahegaonkar) [inline content/python/concepts/statsmodels/terms/glm/glm.md:74]:
```suggestion
import statsmodels.api as sm
from sklearn.datasets import load_iris
import numpy as np

# Load and prepare the iris dataset
iris = load_iris()
X = iris.data[:, [0, 1]]  # using sepal length and sepal width
y = iris.data[:, 2]       # predicting petal length

# Add constant term
X = sm.add_constant(X)

# Fit GLM with Gaussian family (identity link)
model = sm.GLM(y, X, family=sm.families.Gaussian())
results = model.fit()

# Print summary
print(results.summary())

# Make a prediction for a flower with sepal length=5.0 and sepal width=3.5
new_flower = np.array([1, 5.0, 3.5])  # include constant term
new_flower = new_flower.reshape(1, -1)  # Reshape to 2D array for prediction
prediction = results.predict(new_flower)
print("\nPredicted petal length:", prediction[0])
```
`results.predict()` expects a 2D array: The model is trained on a dataset with multiple rows, If we pass a 1D array for new_flower, it will not work because the model expects a 2D array.

File: content/python/concepts/statsmodels/terms/glm/glm.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+# Print model summary
+print("House Price Prediction Model Summary:")
+print(results.summary())
+
+# Make a prediction for a 2000 sq ft house with 3 bedrooms
+new_house = np.array([1, 2000, 3])  # [constant, sq_footage, bedrooms]
+prediction = results.predict(new_house)
+print("\nPredicted price for a 2000 sq ft house with 3 bedrooms:")
+print(f"${prediction[0]:,.2f}k")
```
Comment (SaviDahegaonkar) [inline content/python/concepts/statsmodels/terms/glm/glm.md:138]:
```suggestion
import numpy as np
import statsmodels.api as sm

# Generate sample house data
np.random.seed(42)
n_samples = 100

# Generate features: square footage (1000-3000 sq ft) and bedrooms (2-5)
sq_footage = np.random.uniform(1000, 3000, n_samples)
bedrooms = np.random.randint(2, 6, n_samples)

# Combine features
X = np.column_stack((sq_footage, bedrooms))
X = sm.add_constant(X)  # Add intercept term

# Generate house prices (in thousands)
# Base price: 100k + (200 per sq ft) + (50k per bedroom) + some random noise
y = 100 + 0.2 * X[:, 1] + 50 * X[:, 2] + np.random.normal(0, 25, n_samples)

# Fit GLM with Gaussian family
model = sm.GLM(y, X, family=sm.families.Gaussian())
results = model.fit()

# Print model summary
print("House Price Prediction Model Summary:")
print(results.summary())

# Make a prediction for a 2000 sq ft house with 3 bedrooms
new_house = np.array([1, 2000, 3])  # [constant, sq_footage, bedrooms]
new_house = new_house.reshape(1, -1)  # Reshape to 2D array for prediction
prediction = results.predict(new_house)
print("\nPredicted price for a 2000 sq ft house with 3 bedrooms:")
print(f"${prediction[0]:,.2f}k")
```

**Review by SaviDahegaonkar — APPROVED** [review]
Hey @danialkhan02 , The entry looks good for the second round of review🚀.

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @danialkhan02  😄 

The entry looks good to be merged! 🚀
